spec/models/vm/retirement_management_spec.rb

```diff
@@ -157,4 +157,24 @@
 
     vm.raise_audit_event(event_name, message)
   end
+
+  it "reset retirement state in future" do
+    @vm.update_attributes(:retirement_state => 'retiring')
+    options = { :date => Date.today + 1}
+    @vm.retire(options)
```

Comment:
@gmcculloug 
The retires_on only updates the retires_on attribute and doesn't save the object.